c_binding/lsm_local_disk.c

```diff
@@ -379,6 +482,84 @@ int lsm_local_disk_vpd83_search(const char *vpd83,
     return rc;
 }
 
+int lsm_local_disk_serial_num_get(const char *disk_path, char **serial_num,
+                                  lsm_error **lsm_err)
+{
+    char tmp_serial_num[_LSM_MAX_SERIAL_NUM_LEN];
```

Comment:
`char tmp_serial_num[_LSM_MAX_SERIAL_NUM_LEN];` should be
 `uint8_t tmp_serial_num[_LSM_MAX_SERIAL_NUM_LEN];`.